shared/fs/filepreview/view.js

```diff
@@ -0,0 +1,128 @@
+// @flow
+import * as I from 'immutable'
+import * as React from 'react'
+import * as Styles from '../../styles'
+import * as Types from '../../constants/types/fs'
+import * as Constants from '../../constants/fs'
+import DefaultView from './default-view-container'
+import ImageView from './image-view'
+import TextView from './text-view'
+import AVView from './av-view'
+import * as Kb from '../../common-adapters'
+
+type Props = {|
+  lastModifiedTimestamp: number,
+  mime?: ?Types.Mime,
+  onLoadingStateChange: boolean => void,
+  path: Types.Path,
+  type: Types.PathType,
+  routePath: I.List<string>,
+  url: string,
+|}
+
+type State = {|
+  loadedLastModifiedTimestamp: number,
+|}
+
+export default class extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
```

Comment:
this is likely not what you want. if the other props change you also likely want to reset this etc (like if the file changes). you probably want getDerivedStateFromProps or better componentDidUpdate and reset this if any of the actual data changes (at least path)